### PR TITLE
[MM-41290] Add Endpoint to complete onboarding

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR adds a `/system/onboarding/complete` endpoint. This will be called by the webapp once the onboarding flow in completed and will process the user data. As part of this PR only a list of plugin can be provided, which is then installed and enabled. Additional fields will be added in the future.

The endpoint can be accessed by admins (this is mostly needed for testing, but also to re-run the wizard) and if no user can been created yet.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41290

#### Release Note
```release-note
Added new API endpoint POST /system/onboarding/complete to complete onboarding
```

### CODE REVIEW

File: services/marketplace/client.go [hunk 1]
```diff
@@ -78,6 +88,29 @@ func (c *Client) GetPlugin(filter *model.MarketplacePluginFilter, pluginVersion
 	return nil, errors.New("plugin not found")
 }
 
+func (c *Client) GetLastestPlugin(filter *model.MarketplacePluginFilter) (*model.BaseMarketplacePlugin, error) {
```
Comment (neallred) [inline services/marketplace/client.go:91]: Nit: should this be `GetLatestPlugin` instead of `GetLastestPlugin`?

**Review by neallred — APPROVED** [review]

File: services/marketplace/client.go (hunk 1 above)
Comment (hanzei) [reply services/marketplace/client.go]: Good catch, fixed :+1:

File: i18n/en.json [hunk 2]
```diff
@@ -6055,6 +6055,14 @@
     "id": "app.submit_interactive_dialog.json_error",
     "translation": "Encountered an error encoding JSON for the interactive dialog."
   },
+  {
+    "id": "app.system.complete_onboarding_request.app_error",
+    "translation": "Failed to decode the complete onboarding request."
+  },
+  {
+    "id": "app.system.complete_onboarding_request.no_first_user",
+    "translation": "Onboarding can only be completed by an System Administrator."
```
Comment (neallred) [inline i18n/en.json:6064]: nit: `a System Administrator`

File: api4/system_test.go [hunk 3]
```diff
@@ -778,3 +782,103 @@ func TestPushNotificationAck(t *testing.T) {
...
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+
+		t.Cleanup(func() {
+			resp, err = th.SystemAdminClient.RemovePlugin("testplugin2")
+			require.NoError(t, err)
+			CheckOKStatus(t, resp)
+		})
+
+		installedPlugin, resp, err := th.SystemAdminClient.GetPlugins()
```
Comment (mickmister) [inline api4/system_test.go:871]:
```suggestion
		installedPlugins, resp, err := th.SystemAdminClient.GetPlugins()
```

**Review by mickmister — APPROVED** [review]
LGTM :tada:

Nice work!

**Review by levb — APPROVED** [review]

**Comment by hanzei** [comment]
Discussed with @neallred offline: I'm going to merge this PR to unblock this work on https://github.com/mattermost/mattermost-server/pull/19446. QA will happen as part of the webapp PR testing.

**Pull request merged**